src/app/core/tareas.service.ts

```diff
@@ -30,16 +28,33 @@ export class TareaService {
 
   public addTarea$(tarea: Tarea): Observable<Tarea> {
     const urlPostTarea = this.url + '/tarea';
-    var tareasArrary: Tarea[] = this.tareaBehSub.getValue();
+    let tareasArrary = this.tareaBehSub.getValue();
     tarea.status = TareaStatus.PENDIENTE;
     return this.http.post<Tarea>(urlPostTarea, tarea).pipe(
       catchError(err => { throw new Error("Proceso invalido, problemas con la API-Tareas.") }),
       tap((tarea: Tarea) => {
         tareasArrary.push(tarea);
-        this.tareaBehSub.next(tareasArrary)
+        this.tareaBehSub.next(tareasArrary);
       })
     );
   }
 
+  public editTarea$(tarea: Tarea, id: number): Observable<Tarea> {
+    tarea.id = id;
+    const urlEditTarea = this.url + '/tarea/' + id;
+    //updating array for tareaBehSub 
+    let tareaEditArray = this.updateArray(tarea, id);
+    return this.http.put<Tarea>(urlEditTarea, tarea).pipe(
+      catchError(err => { throw new Error("Proceso invalido, problemas con la API-Tareas.") }),
+      tap(dat => { this.tareaBehSub.next(tareaEditArray) })
+    );
+  }
+
+  updateArray(tarea: Tarea, id: number): Tarea[] {
+    let varArray = this.tareaBehSub.getValue();
```

Comment:
Yo diria de crear un nuevo array en vez de asignar el array asi directo desde el getValue() del behavior, algo como esto: 
`let oldArray = Array.from(this.tareaBehSub.getValue());`
asi si ya estariamos seguros de que es un nuevo array.